src/pages/index.js

```diff
@@ -87,25 +119,92 @@ const addFormValidator = new FormValidator(
   cardForm
 );
 
+const avatarFormValidator = new FormValidator(
+  constants.validationSettings,
+  avatarForm
+);
+
 const profileInfo = new UserInfo({
   name: ".profile__name",
   description: ".profile__description",
 });
 
-const editProfilePopup = new PopupWithForm("#edit-popup", handleSaveProfile);
-const newCardPopup = new PopupWithForm("#new-card-popup", handleAddCard);
+const editProfilePopup = new PopupWithForm("#edit-popup", handleUpdateProfile);
+
+const updateAvatarPopup = new PopupWithForm(
+  "#avatar-popup",
+  handleUpdateAvatar
+);
+
 const imagePopup = new PopupWithImage("#preview-image-popup");
 
 editFormValidator.enableValidation();
 addFormValidator.enableValidation();
+avatarFormValidator.enableValidation();
 
 // ! ||--------------------------------------------------------------------------------||
 // ! ||                                Gallery Creation                                ||
 // ! ||--------------------------------------------------------------------------------||
 
-const cardGallery = new Section(
-  { items: constants.initialCards, renderer: renderCard },
-  ".gallery__cards"
-);
-
-cardGallery.renderItems();
+api.getInitialCards().then((cards) => {
+  const cardGallery = new Section(
+    { items: cards, renderer: renderCard },
+    ".gallery__cards"
+  );
+
+  function renderCard(cardData) {
+    const card = new Card(
+      {
+        cardData,
+
+        handleImageClick,
+        handleDeleteButton,
+        handleCardLike,
+        handleCardUnlike,
+      },
+      cardTemplate
+    );
+
```

Comment:
Needs correction: the functions from lines 168-191 should be declared in the global scope